### PR TITLE
Laravel migrations

### PR DESCRIPTION
**Pull request by pheinrichs**
This branch is functional currently but the validation errors are through the roof when using the migrations.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: database/migrations/2018_07_03_091314_create_pollers_table.php [hunk 1]
```diff
@@ -3,34 +3,37 @@
...
+            $table->integer('id');
+            $table->string('poller_name');
+            $table->dateTime('last_polled');
+            $table->integer('devices');
+            $table->float('time_taken', 10, 0);
+        });
 
+        \DB::statement("ALTER TABLE `pollers` ADD UNIQUE `id` (`id`);");
+        \DB::statement("ALTER TABLE `pollers` CHANGE `id` `id` int(11) NOT NULL auto_increment;");
+        \DB::statement("ALTER TABLE `pollers` ADD PRIMARY KEY (`poller_name`);");
```
Comment (pheinrichs) [inline database/migrations/2018_07_03_091314_create_pollers_table.php]: I've fixed all the validation issues as of now except for this file. The Validation gives this error. 
```
ALTER TABLE `pollers` DROP INDEX `id`, INDEX `id` (`id`);
```
When attempting to add this, it breaks the migration with this error
```
SQLSTATE[42000]: Syntax error or access violation: 1064 
```

Any thoughts on why?

**Comment by murrant** [comment]
Need to update daily.php and the install process too.

File: database/migrations/2018_07_03_091314_create_access_points_table.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            $table->tinyInteger('radioutil')->default(0);
+            $table->smallInteger('numasoclients')->default(0);
+            $table->smallInteger('nummonclients')->default(0);
+            $table->tinyInteger('numactbssid')->default(0);
+            $table->tinyInteger('nummonbssid')->default(0);
+            $table->unsignedTinyInteger('interference');
+            $table->index(['name','radio_number'], 'name');
+        });
+
+        \DB::statement("ALTER TABLE `access_points` CHANGE `channel` `channel` tinyint(4) unsigned NOT NULL DEFAULT '0' ;");
```
Comment (laf) [inline database/migrations/2018_07_03_091314_create_access_points_table.php]: How come this is needed?

File: database/migrations/2018_07_03_091314_create_alert_log_table.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        Schema::create('alert_log', function (Blueprint $table) {
+            $table->integer('id', true);
+            $table->integer('rule_id')->index('rule_id');
+            $table->integer('device_id')->index('device_id');
+            $table->integer('state');
+            $table->binary('details')->nullable();
+            $table->timestamp('time_logged')->default(DB::raw('CURRENT_TIMESTAMP'))->index('time_logged');
+        });
+
+        \DB::statement("ALTER TABLE `alert_log` CHANGE `details` `details` longblob NULL ;");
```
Comment (laf) [inline database/migrations/2018_07_03_091314_create_alert_log_table.php]: Same for this one?

File: database/migrations/2018_07_03_091314_create_alerts_table.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            $table->integer('rule_id')->index('rule_id');
+            $table->integer('state');
+            $table->integer('alerted');
+            $table->integer('open');
+            $table->text('note', 65535)->nullable();
+            $table->timestamp('timestamp')->default(DB::raw('CURRENT_TIMESTAMP'));
+            $table->unique(['device_id','rule_id'], 'unique_alert');
+        });
+
+        \DB::statement("ALTER TABLE `alerts` CHANGE `timestamp` `timestamp` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP on update CURRENT_TIMESTAMP;");
```
Comment (laf) [inline database/migrations/2018_07_03_091314_create_alerts_table.php]: Same here.

**Comment by pheinrichs** [comment]
@laf the reason for the raw schema changes are to pass the database validation test. Our previous migrations are a little different then the Laravel ones so without the raw changes, causes the validate to spit out errors.

IMO, for any database insertions. I’d like to keep them contained strictly in the migration because it would make it easy. I just had the seeds temporarily. Open for discussion though I guess.

**Comment by laf** [comment]
> IMO, for any database insertions. I’d like to keep them contained strictly in the migration because it would make it easy. I just had the seeds temporarily. Open for discussion though I guess.

Ok so in essence the same is it is now but obviously migrated to migrations.

**Comment by laf** [comment]
I get this when trying to run install.php, the db schema creation has a laravel error and this is in librenms.log:

```
[2018-10-20 22:04:42] production.ERROR: ErrorException: ini_set(): A session is active. You cannot change the session module's ini settings at this time in /opt/librenms/LibreNMS/Config.php:433 Stack trace: #0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'ini_set(): A se...', '/opt/librenms/L...', 433, Array) #1 /opt/librenms/LibreNMS/Config.php(433): ini_set('session.cookie_...', '1') #2 /opt/librenms/LibreNMS/Config.php(57): LibreNMS\Config::processConfig(false) #3 /opt/librenms/app/Providers/AppServiceProvider.php(29): LibreNMS\Config::load() #4 [internal function]: App\Providers\AppServiceProvider->boot() #5 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array) #6 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}() #7 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure)) #8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL) #9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(788): Illuminate\Container\Container->call(Array) #10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(771): Illuminate\Foundation\Application->bootProvider(Object(App\Providers\AppServiceProvider)) #11 [internal function]: Illuminate\Foundation\Application->Illuminate\Foundation\{closure}(Object(App\Providers\AppServiceProvider), 15) #12 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(772): array_walk(Array, Object(Closure)) #13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\Foundation\Application->boot() #14 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(208): Illuminate\Foundation\Bootstrap\BootProviders->bootstrap(Object(Illuminate\Foundation\Application)) #15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(267): Illuminate\Foundation\Application->bootstrapWith(Array) #16 /opt/librenms/LibreNMS/Util/Laravel.php(47): Illuminate\Foundation\Console\Kernel->bootstrap() #17 /opt/librenms/includes/init.php(96): LibreNMS\Util\Laravel::bootCli() #18 /opt/librenms/html/includes/output/db-update.inc.php(31): require('/opt/librenms/i...') #19 /opt/librenms/html/ajax_output.php(35): require('/opt/librenms/h...') #20 {main} [] []
```

**Comment by laf** [comment]
I've moved this to 1.46, I think it's a bit too much to get in for 1.45 which is only 1 week away.

**Comment by murrant** [comment]
Yeah, a few docs updated needed.  Both ./build-base.php and includes/sql-schema/update.php invoke artisan migrate if dbSchema is 1000 or non-existent.

**Pull request merged**